### PR TITLE
Two-pane UI on large screens

### LINKED ISSUE #8168: Two-pane Status UI on large screens
### User Story

As a user, I want to have two-pane UI on large screens (tabletы f.ex.) so that I don't have to spend time navigating back and forth from messages list to chat list.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: On the big screens, when space allows it would be more convenient to use two-pane UI. We already have a design for it - https://www.figma.com/file/YS5wVm8MDFvbmXROj1oMZmM4/Tablet-Desktop?node-id=1%3A2

#### Expected behavior
When space allows, Status UI become two-pane to fit more information for user
[comment]: # (Describe what you expected to happen.)

#### Actual behavior
Currently even on a big tablet screens user see only one pane with info. That makes navigating more tedious.

[comment]: # (Describe what actually happened.)

### PR DESCRIPTION
**Pull request by vkjr**
fixes #8168

### Summary

UI switched to 2-pane format when device width is >640px.

### Review notes
2-pane ui navigator implemented here:
https://github.com/status-im/react-native-navigation-twopane

### Testing notes
Per discussion with @mandrigin: it would be good to merge 2-pane ui even if there are some discrepancies with designs. They can be solved as a separate issues.

#### Platforms

- Android
- iOS

#### Areas that maybe impacted
Navigation

##### Functional

- 1-1 chats
- public chats
- group chats
- wallet / transactions
- dapps / app browsing
- account recovery
- new account

### Steps to test

- Open Status
- check that it shows 2-pane UI when device width bigger than 640px
- check that navigation between chats and inside right pane works correclty

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/routing/core.cljs [hunk 1]
```diff
@@ -135,6 +135,27 @@
                                        (bottom-bar/minimize-bar route-name)))))}
            (prepare-config config)))))
 
+(defn twopane-navigator [routes config]
+  (navigation/twopane-navigator
+   routes
+   (cond->
```
Comment (cammellos) [inline src/status_im/ui/screens/routing/core.cljs]: not sure I get this, mabye some leftover? You are not threading the value, so `cond->` is unnecessary looks like

**Review by cammellos — APPROVED** [review]

File: src/status_im/ui/screens/views.cljs [hunk 2]
```diff
@@ -53,6 +55,31 @@
...
+(defn dimensions-fit-two-pane? [dimensions]
+  (let [width (get dimensions :width)]
+    (>= width constants/two-pane-min-width)))
+
+(defn reset-component-on-mount [view-id component two-pane?]
+  (when (and @initial-view-id
+             (or
+              js/goog.DEBUG
+              (not @component)))
+    (do
```
Comment (rasom) [inline src/status_im/ui/screens/views.cljs]: `do` is not needed here

File: src/status_im/ui/screens/views.cljs [hunk 3]
```diff
@@ -53,6 +55,31 @@
...
+      (reset! component (routing/get-main-component
+                         (if js/goog.DEBUG
+                           @initial-view-id
+                           @view-id)
+                         two-pane?)))))
+
+(defn reset-component-on-update [view-id component two-pane?]
+  (when (and @initial-view-id (not @component))
+    (do
```
Comment (rasom) [inline src/status_im/ui/screens/views.cljs]: `do` is not needed

**Review by rasom — APPROVED** [review]

**Comment by hesterbruikman** [comment]
Is anyone testing on an Android tablet? Would greatly appreciate a recording 🙏

**Comment by errorists** [comment]
👋 I could test it on an iPad if someone would add my device's UDID to provisioning profile.

File: src/status_im/ui/screens/routing/core.cljs (hunk 1 above)
Comment (vkjr) [reply src/status_im/ui/screens/routing/core.cljs]: I created `twopane-navigator` by example of `stack-navigator` from @rasom. 
@rasom, could you please take a look, do we still need `cond->` here? Thanks.

**Comment by annadanchenko** [comment]
real device ipad mini 3, iOS 12.1. Tested ok:
- start new chat -> scan QR code to add a contact
- scan Qr code (also scanned contact code of a user)
- Wallet -> send -> select recipient -> scan QR code of valid address
- Profile -> Edit -> Capture a photo for profile picture
- enable "Use biometric authentication" and can use touch id to log in
- going to offline state and back to online

**Comment by churik** [comment]
@vkjr 
Some findings:
### 1) On android after joining to public chat input in `topic` is not erased

inp-l

### 2) Unread messages are not marked as read 
**Steps:**
- create account
- join to some public chats
- send to some public chat messages from another device to see unread badge
- switch to chat with unread badge, scroll
- switch to another chat
**Expected result:** unread messages badge is not shown
**Actual result:**  unread messages badge is preserved
**Log:** [StatusIm-190624-072703-cb2498-pr.ipa_2019_6_25_10_25_38.log](https://github.com/status-im/status-react/files/3324889/StatusIm-190624-072703-cb2498-pr.ipa_2019_6_25_10_25_38.log)

After relogin all messages are marked as read.
Probably depending on speed of switching between chats, I could reproduce 3/5 times.
ezgif com-video-to-gif (2)

### 3) Tapping several times on back button leads to white screen (IOS)
 **Steps:**
- create account
- click on + > start new chat 
- tap on back 2 times
 **Expected result:** can go back after 1 tap
 **Actual result:** white screen
**Device: ** IPad Air
**IOS:** 12.3..1
### 4) System messages in chat are not visible
photo_2019-06-25 13 23 13
ssdfsd

Also fetching animation is shown  on two panels:
Monosnap 2019-06-25 13-31-34

And probably can be fixed later:
### 4) Recently used stickers are overlapped
Stickers

**Comment by errorists** [comment]
a small but important improvement if we could add it here, a 1px Light Grey dividing line between the two columns so they're visibly separated, similar to current Desktop.

**Comment by vkjr** [comment]
Thanks for findings! Will check them.

**Comment by flexsurfer** [comment]
@vkjr any updates?

**Comment by vkjr** [comment]
@flexsurfer, I fixed most of the issues but there is one that I'm a bit stuck on - marking messages as "read" when you navigate to chat. 
In default mobile ui messages marked read in a response for navigation`onWillMount` event. But 2-pane navigator doesn't send that event.
I tried to fix that from navigator side but had no luck so far. Currently I think that maybe this behavior is correct because in fact chat page doesn't unmount in 2-pane ui when you switch between chats.
So probably I'll just move marking messages as read from `onWillMount` to function where user switches between chats, wdyt?

**Comment by flexsurfer** [comment]
@vkjr i believe it's even better, people hate these unread badges, @hesterbruikman wdyt ?

**Comment by errorists** [comment]
@flexsurfer dunno, I wouldn't call it better, I'd say we're waiting for better to come with this issue :) https://github.com/status-im/status-react/issues/6375

**Comment by flexsurfer** [comment]
@errorists yes, thanks :)

**Comment by bitsikka** [comment]
> @flexsurfer dunno, I wouldn't call it better, I'd say we're waiting for better to come with this issue :) #6375

👍 
I'm working on #6375; particularly new unread badge thingy right now :)

this issue/feature does invalidate few of the old behaviors including chats tab badge

**Comment by vkjr** [comment]
@churik, all issues fixed!

**Comment by asemiankevich** [comment]
@vkjr please resolve conflicts here.

Also, last message in public chat is hidded and not visible (Galaxy Tab 4)

image

And the issue with topic is not erased is still in place

**Comment by vkjr** [comment]
@asemiankevich, thanks for testing! Will check.

**Comment by asemiankevich** [comment]
also on iPad Pro 3d gen there is no any delimiter between panes, can we add one?
image

**Comment by asemiankevich** [comment]
@vkjr as we discussed please resolve conflicts and rebase, then we can run e2e and merge, the issues i mentioned we can fix separately

**Comment by vkjr** [comment]
@asemiankevich, ready, please check

**Comment by asemiankevich** [comment]
lets merge this one and i will address the issues separately, thanks for your patience and sorry for delays! @vkjr

**Comment by vkjr** [comment]
@asemiankevich, thanks, merged!

**Comment by ghost** [comment]
`desktop_files/yarn.lock` and `mobile_files/yarn.lock` changed. Pinging @jakubgs, @PombeirP, and @corpetty

**Pull request merged**